### PR TITLE
EFM32 - change bitfield definitions to be unshifted.

### PR DESCRIPTION
**Pull request by kbob**
Change all bitfield constant definitions for EFM32 to use unshifted values.  Also fixed three errata in the definitions.

I also have an update for [libopencm3-examples pull request #85](https://github.com/libopencm3/libopencm3-examples/pull/85) that uses the updated definitions.  I will submit a pull request for that momentarily.

### Rationale

In the comments on Pull Request #581, @karlp and @ChuckM asked that the bitfield definitions be unshifted.  E.g., instead of
```c
#define REG_FIELD_SHIFT 2
#define REG_FIELD_MASK 0xC0
#define REG_FIELD(x) (((x) << REG_FIELD_SHIFT) & REG_FIELD_MASK)
#define REG_FIELD_RED REG_FIELD(1)
#define REG_FIELD_BLUE REG_FIELD(2)

   REG = REG_FIELD_RED;
```
this is the preferred way.
```c
#define REG_FIELD_RED 1
#define REG_FIELD_BLUE 2

    REG = REG_FIELD(REG_FIELD_RED);
```

### CODE REVIEW

**Comment by kbob** [comment]
The libopencm3-examples pull request is [#116](https://github.com/libopencm3/libopencm3-examples/pull/116).

File: include/libopencm3/efm32/lg/acmp.h [hunk 1]
```diff
@@ -48,28 +48,28 @@
...
-#define ACMP_CTRL_WARMTIME_256CYCLES	ACMP_CTRL_WARMTIME(6)
-#define ACMP_CTRL_WARMTIME_512CYCLES	ACMP_CTRL_WARMTIME(7)
+#define ACMP_CTRL_WARMTIME_4CYCLES	(0)
+#define ACMP_CTRL_WARMTIME_8CYCLES	(1)
+#define ACMP_CTRL_WARMTIME_16CYCLES	(2)
+#define ACMP_CTRL_WARMTIME_32CYCLES	(3)
+#define ACMP_CTRL_WARMTIME_64CYCLES	(4)
+#define ACMP_CTRL_WARMTIME_128CYCLES	(5)
+#define ACMP_CTRL_WARMTIME_256CYCLES	(6)
+#define ACMP_CTRL_WARMTIME_512CYCLES	(7)
```
Comment (ChuckM) [inline include/libopencm3/efm32/lg/acmp.h]: When you are #defining simply a number do you really need parenthesis around it?

**Comment by ChuckM** [comment]
other than the admittedly harmless extra parenthsis it looks good to me @kbob

**Comment by kbob** [comment]
I removed the parentheses.

**Comment by kbob** [comment]
This is superceded by #606.

**Pull request closed without merging**